### ISSUE TITLE
TCP Microservice - Validation

### ISSUE THREAD

**Issue by affanshahid** [nestjs/nest#3114]
Validation doesn't seem to be working with a TCP transport and a hybrid app.

My controller:
```ts
@UsePipes(new ValidationPipe())
@Controller()
export class NotificationsController {
  @MessagePattern('configuration-change')
  accumulate(@Body(new ValidationPipe()) configurationChangeDto: ConfigurationChangeDto): Observable<number> {
    console.log(configurationChangeDto);
    return from([1, 2, 3]);
  }
}
```

My DTO:
```ts
import { IsNotEmpty, IsUUID, IsString } from 'class-validator';

export class ConfigurationChangeDto {
  @IsNotEmpty()
  @IsUUID("4")
  configurationId!: string;

  @IsNotEmpty()
  @IsString()
  path!: string;
}
```

However I get no error when I call this endpoint using an array of numbers.

Nest version: 6.8.2

EDIT: Actually I can't even get a custom pipe to run, the transform method isn't called. Am I doing something wrong?

**Comment by joeyslack** [comment]
You don't need the `@UsePipes()` since you're instantiating at method level.

or

If you keep `@UsePipes()`, then make your body like `@Body()` (`new ValidationPipe()) not needed.

https://docs.nestjs.com/pipes#binding-pipes

**Comment by affanshahid** [comment]
I added it there since the one at method level was not working. Just to be sure

**Comment by kamilmysliwiec** [comment]
> @Body(new ValidationPipe())

`@Body()` shouldn't be used for microservices. 

Can you share a repository which reproduces your issue?

**Comment by kamilmysliwiec** [comment]
Fixed in 6.8.3. Please, let me know if you face any issue.
